hypershift-operator/metrics.go

```diff
@@ -178,7 +202,36 @@ func (m *hypershiftMetrics) observeHostedClusters(hostedClusters *hyperv1.Hosted
 	hcCount := newLabelCounter()
 	hcByConditions := newLabelCounter()
 
+	identityProvidersCounter := make(map[string]float64)
 	for _, hc := range hostedClusters.Items {
+		// Collect proxy metric.
+		var proxyHTTP, proxyHTTPS, proxyTrustedCA string
+		if hc.Spec.Configuration != nil && hc.Spec.Configuration.Proxy != nil {
+			if hc.Spec.Configuration.Proxy.HTTPProxy != "" {
+				proxyHTTP = "1"
+			}
+			if hc.Spec.Configuration.Proxy.HTTPSProxy != "" {
+				proxyHTTPS = "1"
+			}
+			if hc.Spec.Configuration.Proxy.TrustedCA.Name != "" {
+				proxyTrustedCA = "1"
+			}
+			m.clusterProxy.WithLabelValues(hc.Namespace, hc.Name, proxyHTTP, proxyHTTPS, proxyTrustedCA).Set(1)
+		}
+
+		// Group identityProviders by type.
+		if hc.Spec.Configuration != nil && hc.Spec.Configuration.OAuth != nil {
+			for _, identityProvider := range hc.Spec.Configuration.OAuth.IdentityProviders {
+				identityProviderType := string(identityProvider.Type)
+				identityProvidersCounter[identityProviderType] = identityProvidersCounter[identityProviderType] + 1
+			}
+		}
+
+		// Collect limited support metric.
+		if _, ok := hc.Labels[hyperv1.LimitedSupportLabel]; ok {
+			m.clusterProxy.WithLabelValues(hc.Namespace, hc.Name).Set(1)
+		}
```

Comment:
set the metric value to `0` if label not present
https://github.com/openshift/osd-metrics-exporter/blob/be64e94da0caa0ba7941324084b4b14ba97c70c6/pkg/metrics/aggregator.go#L161-L173